### PR TITLE
tidb_query: Implement collation aware StreamAggr

### PR DESCRIPTION
**Pull request by andylokandy**
Signed-off-by: Dian Luo <andylokandy@hotmail.com>

###  What have you changed?

Implement collation aware StreamAggr

###  What is the type of the changes?

- New feature (a change which adds functionality)
###  How is the PR tested?

- Unit test
- Integration test

### CODE REVIEW

File: components/tidb_query/src/batch/executors/stream_aggr_executor.rs [hunk 1]
```diff
@@ -247,30 +254,47 @@ impl<Src: BatchExecutor> AggregationExecutorImpl<Src> for BatchStreamAggregation
...
-                    if logical_row_idx > 0 {
-                        update_current_states(
-                            context,
-                            &mut self.states,
-                            aggr_fn_len,
-                            &self.aggr_expr_results_unsafe,
-                            group_start_logical_row,
-                            logical_row_idx,
-                        )?;
+            let group_match = 'group_match: {
```
Comment (sticnarf) [inline components/tidb_query/src/batch/executors/stream_aggr_executor.rs]: I think a closure + return can avoid using this unstable feature.

Comment (andylokandy) [reply components/tidb_query/src/batch/executors/stream_aggr_executor.rs]: done

**Pull request merged**